modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -351,8 +351,15 @@ INSFVRhieChowInterpolator::initialize()
 
   // Reset map of coefficients to zero.
   // The keys should not have changed unless the mesh has changed
-  for (const auto & pair : _a)
-    _a[pair.first] = 0;
+  // Dont reset if not in current system
+  // IDEA: clear them derivatives
+  if (_u->sys().number() == _fe_problem.currentNlSysNum())
+    for (const auto & pair : _a)
+      _a[pair.first] = 0;
+  else
+    for (const auto & pair : _a)
+      _a[pair.first] = {
+          _a[pair.first](0).value(), _a[pair.first](1).value(), _a[pair.first](2).value()};
```

Comment:
The guy who made this code was really dumb (it was me)... I am iterating over the key-value pairs and then I do a lookup with the key when I already have access to the value?! This code can definitely be improved in multiple ways. I will put up a PR

Reply:
interesting that we 0 in one case and set the raw value in another. 
did I do that part?

Reply:
lol in the PR you made that got merged two hours ago? Yes in this PR you added the raw value part (without using `raw_value` which #25898 will switch to).

Reply:
Yeah. Seems that PR was wrong

Reply:
I'm confused. You're saying that the *this* PR is wrong?

Reply:
Yeah I think so. 
Probably bad copy paste I ll check when I have a machine ready

Reply:
nevermind it s correct